sql_assign2.sql

```diff
@@ -0,0 +1,183 @@
+-- For this assignment, assume you have a database with the following table:
+-- CREATE TABLE Employees (
+--     EmployeeID INT PRIMARY KEY,
+--     FirstName VARCHAR(50),
+--     LastName VARCHAR(50),
+--     Department VARCHAR(50),
+--     Salary DECIMAL(10, 2),
+--     HireDate DATE
+-- );
+
+CREATE DATABASE Test;
+
+USE Test;
+
+CREATE TABLE Employees (
+    EmployeeID INT PRIMARY KEY,
+    FirstName VARCHAR(50),
+    LastName VARCHAR(50),
+    Department VARCHAR(50),
+    Salary DECIMAL(10, 2),
+    HireDate DATE
+);
+
+INSERT INTO Employees (EmployeeID, FirstName, LastName, Department, Salary, HireDate) VALUES
+(1, 'John', 'Doe', 'Sales', 50000.00, '2022-01-15'),
+(2, 'Jane', 'Smith', 'Marketing', 60000.00, '2023-02-20'),
+(3, 'Michael', 'Johnson', 'HR', 55000.00, '2021-03-10'),
+(4, 'Emily', 'Davis', 'Finance', 70000.00, '2020-04-25'),
+(5, 'Daniel', 'Wilson', 'IT', 75000.00, '2019-05-30'),
+(6, 'Olivia', 'Brown', 'Sales', 48000.00, '2022-06-14'),
+(7, 'James', 'Jones', 'Marketing', 62000.00, '2023-07-19'),
+(8, 'Sophia', 'Garcia', 'HR', 53000.00, '2021-08-22'),
+(9, 'Liam', 'Martinez', 'Finance', 80000.00, '2020-09-05'),
+(10, 'Isabella', 'Hernandez', 'IT', 72000.00, '2019-10-12');
+
+
+-- Create a table students and insert names in malayalam
+CREATE TABLE Students (Stdid int, Stdname varchar(30));
+Alter Table Students Alter column Stdname nvarchar(30);
+
+INSERT INTO Students values (101, 'അജിത്');
+INSERT INTO Students values (102, 'ബിജു');
+INSERT INTO Students values (103, 'സ്നേഹ');
+INSERT INTO Students values (104, 'ശ്രുതി');
+
+-- Retrieve all employees who work in Sales, Marketing, or IT departments.
+SELECT * FROM Employees WHERE Department IN ('Sales','Marketing','IT') 
+
+
+-- Find all employees with salaries ranging from $50,000 to $75,000 (inclusive).
+SELECT * FROM Employees WHERE Salary BETWEEN 50000 AND  75000
+
+
+-- List all employees whose last name begins with the letter 'S'.
+SELECT * FROM Employees WHERE LastName LIKE 'S%';
+
+-- Display all employees with exactly five letters in their first name.
+SELECT * FROM Employees WHERE FirstName LIKE '_____'
+
+-- Find employees whose last name starts with either 'B', 'R', or 'S'.
+SELECT * FROM Employees WHERE LastName LIKE 'B%' OR  LastName LIKE 'R%' OR LastName LIKE 'S%';
+SELECT * FROM Employees WHERE LastName LIKE '[B,R,S]%'
+
+
+-- Retrieve all employees whose first name begins with any letter from 'A' through 'M'.
+SELECT * FROM Employees WHERE FirstName BETWEEN 'A' AND 'M'
+
+-- List employees whose last name doesn't start with a vowel (A, E, I, O, U).
+SELECT * FROM Employees WHERE LastName LIKE '[^A,E,I,O,U]%'
+
+-- Identify employees earning more than $80,000 annually. 
+SELECT * FROM Employees WHERE Salary > 80000
+
+-- Find employees who joined the company before 2020.
+SELECT * FROM Employees WHERE HireDate < '2020-01-01'
+SELECT * FROM Employees WHERE HireDate < '2020'
```

Comment:
Will this work ?

Reply:
![Screenshot 2024-10-03 173320](https://github.com/user-attachments/assets/2956e1f0-657f-4f12-a151-ddca97fc6141)